pyccel/ast/numpyext.py

```diff
@@ -511,23 +507,19 @@ def _sympystr(self, printer):
     def fprint(self, printer, lhs=None):
         """Fortran print."""
 
-        init_value = '[({0} + {1}*{2},{1} = 0,{3}-1)]'
-
-        start = printer(self.start)
-        step  = printer(self.step)
-        stop  = printer(self.stop)
-        index = printer(self.index)
-
-        init_value = init_value.format(start, index, step, stop)
-
+        template = '[({start} + {index}*{step},{index} = 0,{stop}-1)]'
```

Comment:
Shouldn't we use `PyccelMinus` to do the `-1`? That way it is printed with its precision like all the other integers

Reply:
Yes, you're right.

I've copied this line from somewhere else without thinking much about it. I should track it down and change that too.

Reply:
By the way, shouldn't we use the correct precision also to print `0`?

Reply:
Yes that would be better. If 0 is stored as in the `Integer` class then it would be printed correctly (or at least with the default precision)

Reply:
Done in 9aed2d3